src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -907,5 +908,20 @@ public static void SkipInit<T>(out T value)
             // unbox !!T
             // ret
         }
+
+
+        // Internal helper methods:
+
+        // Determines if the address is aligned at least to `alignment` bytes.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsOpportunisticallyAligned<T>(ref readonly T address, nuint alignment)
```

Comment:
There are more places in CoreLib that should use this method.

Reply:
I've only done System.Memory so far, as suggested by Tanner (to break it up by area). Did you want me to do the whole repo (this would be my preferred option), or a larger portion at once (let me know how much if this)?

Reply:
I have seen ~5 places where this helper can be used. I think all of them can be fixed up in this PR.

Outside CoreLib, I have only seen one hit System.Text.Json where the shared helper cannot be used without introducing a lot of complexity.

Reply:
So, update the other places to use alignment helper, but update the rest of the code in them separately? Or just do all the core library code?

Reply:
> as suggested by Tanner (to break it up by area, or at least that's how I understood it)

I mostly meant this as a way to help scope the changes based on relation to eachother. If the PR contains changes to S.T.Encoding and S.T.Json for example, it might be tougher to get in.

In this case, you're introducing a new API and updating places to use it en masse, so it's fine since that's all directly related to the new internal API, especially given there's only 4-5 places.